com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StepRequestHandler.java

```diff
@@ -438,15 +439,19 @@ private boolean shouldDoExtraStepInto(int originalStackDepth, Location originalL
         return true;
     }
 
-    private boolean isSameLocation(Location original, Location current) {
+    private boolean isSameLocation(Location current, Location original, MethodInvocation targetStepIn) {
         if (original == null || current == null) {
             return false;
         }
 
         Method originalMethod = original.method();
         Method currentMethod = current.method();
+        // if the lines doesn't match, check if the current line is still behind the
+        // target if a target exist. This handles where the target is part of a
+        // expression which is wrapped.
         return originalMethod.equals(currentMethod)
-            && original.lineNumber() == current.lineNumber();
+                && (original.lineNumber() == current.lineNumber()
+                        || (targetStepIn != null && targetStepIn.lineStart >= current.lineNumber()));
```

Comment:
this doesn't work for multiline method invocation. We should use `targetStepIn.lineEnd` in the last check, e.g. `targetStepIn.lineEnd >= current.lineNumber()`.

For the use case below, step-into target `filterMe` doesn't work, it stops at the line of `"2", "3", `.
```java
new ArrayList<>(List.of("1"))
        .subList(0, 1)
        .addAll(filterMe("1", 
            "2", "3", 
            "4"));
```